src/main/java/net/imglib2/algorithm/hough/HoughTransforms.java

```diff
@@ -0,0 +1,446 @@
+	 * </pre>
+	 * <p>
+	 * In other words, {@code rho} represents the signed minimum distance from
+	 * the image origin to the line, and {@code theta} indicates the angle
+	 * between the row-axis and the minimum offset vector.
+	 * </p>
+	 * <p>
+	 * For a given point, then, votes are placed along the curve
+	 * </p>
+	 *
+	 * <pre>
+	 * rho = y * sin( theta ) + x * cos( theta )
+	 * </pre>
+	 * <p>
+	 * It is important to note that the interval of the first dimension of the
+	 * vote space image is NOT {@code [-maxRho, maxRho)} but instead
+	 * {@code [0, maxRho * 2)}; the same applies to the second dimension of the
+	 * vote space as well. Thus if {@link HoughTransforms#pickPeaks} is not used
+	 * to retrieve the maxima from the vote space, the vote space will have to
+	 * be translated by {@code -maxRho} in dimension 0 to get the correct
+	 * {@code rho} and {@code theta} values from the vote space.
+	 * </p>
+	 *
+	 * @param input
+	 *            - the {@link RandomAccessibleInterval} to run the Hough Line
+	 *            Transform over
+	 * @param votespace
+	 *            - the {@link RandomAccessibleInterval} in which the results
+	 *            are stored
+	 * @param nTheta
+	 *            - the number of bins for theta resolution
+	 * @param nRho
+	 *            - the number of bins for rho resolution
+	 * @param filter
+	 *            - a {@link Predicate} judging whether or not the a value is
+	 *            above the minimum value allowed by the populator. Any input
+	 *            less than or equal to this value will be disregarded by the
+	 *            populator.
+	 */
+	public static < T, U extends IntegerType< U > > void voteLines(
+			final RandomAccessibleInterval< T > input,
+			final RandomAccessibleInterval< U > votespace,
+			final int nTheta,
+			final int nRho,
+			final Predicate< T > filter )
+	{
+
+		final long[] dims = new long[ input.numDimensions() ];
+		input.dimensions( dims );
+
+		final double minRho = -computeLength( dims );
+		final double dRho = 2 * computeLength( dims ) / nRho;
+
+		final double minTheta = -Math.PI / 2;
+		final double dTheta = Math.PI / nTheta;
+
+		final double[] rho = new double[ nRho ];
+		final double[] theta = new double[ nTheta ];
+
+		// create theta LUT
+		for ( int t = 0; t < nTheta; ++t )
+		{
+			theta[ t ] = dTheta * t + minTheta;
+		}
+
+		// create rho LUT
+		for ( int r = 0; r < nRho; ++r )
+		{
+			rho[ r ] = dRho * r + minRho;
+		}
+
+		final Cursor< T > imageCursor = Views.iterable( input ).localizingCursor();
+		final RandomAccess< U > outputRA = votespace.randomAccess();
+
+		while ( imageCursor.hasNext() )
+		{
+			double fRho;
+			int r;
+
+			imageCursor.fwd();
+
+			if ( filter.test( imageCursor.get() ) )
+			{
+				for ( int t = 0; t < nTheta; ++t )
+				{
+					fRho = Math.cos( theta[ t ] ) * imageCursor.getDoublePosition( 0 ) + Math.sin( theta[ t ] ) * imageCursor.getDoublePosition( 1 );
+					r = Math.round( ( float ) ( ( fRho - minRho ) / dRho ) );
```

Comment:
- Do not cast to `float` before rounding. 
- Is rounding correct? That would mean that the value of each `r` bin defines the center of said bin, not the min. If that is the desired behavior, ignore this second bullet point.

Reply:
@hanslovsky you are right about the cast. I do however think that this rounding is correct since we are using `r` to approximate our value of `rho`, meaning that we want to round up in the case that `rho` is closer to the `r` greater than it than it is to the `r` less than it, if that makes sense.